### PR TITLE
[argon2-browser] Create typings

### PR DESCRIPTION
**Pull request by ivangabriele**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by ivangabriele** [comment]
Just fyi, I added a namespace because we indeed need to call `import { argon2 } from 'argon2-browser'` to be able to use its properties (`argon2.hash`, `argon2.ArgonType`).

File: types/argon2-browser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        hash: Uint8Array;
+        hashHex: string;
+    }
+
+    export interface ArgonType {
+        Argon2d: 0;
+        Argon2i: 1;
+    }
+
+    export namespace ArgonType {
```
Comment (andy-ms) [inline types/argon2-browser/index.d.ts]: Maybe this should be a `const enum`?

File: yargs.d.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+/** Declaration file generated by dts-gen */
```
Comment (andy-ms) [inline yargs.d.ts]: Remove unrelated changes.

File: types/argon2-browser/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Type definitions for argon2-browser 1.1
+// Project: https://github.com/antelle/argon2-browser#readme
+// Definitions by: Ivan Gabriele <https://github.com/ivangabriele>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export namespace argon2 {
```
Comment (andy-ms) [inline types/argon2-browser/index.d.ts]: Seems wierd for a library to expose itself via a single export, but that's apparently what it does.

File: types/argon2-browser/argon2-browser-tests.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { argon2 } from 'argon2-browser';
+
+const mandatoryOptions = {
+  pass: 'Qwerty12?',
+  salt: 'Salty'
+};
+
+argon2.hash(mandatoryOptions).then(result => result.encoded); // $ExpectType Promise<string>
```
Comment (andy-ms) [inline types/argon2-browser/argon2-browser-tests.ts]: Might be easier to test inside of an async function, so you could do:
```ts
(await argon2.hash(mandatoryOptions)).encoded; // $ExpectType string
```

File: types/argon2-browser/index.d.ts (hunk 3 above)
Comment (ivangabriele) [reply types/argon2-browser/index.d.ts]: Indeed...

**Comment by ivangabriele** [comment]
Thanks for the comments, that should be better now !

**Comment by andy-ms** [comment]
Thanks!
